src/llfs/track_fds.cpp

```diff
@@ -0,0 +1,188 @@
+//#=##=##=#==#=#==#===#+==#+==========+==+=+=+=+=+=++=+++=+++++=-++++=-+++++++++++
+//
+// Part of the LLFS Project, under Apache License v2.0.
+// See https://www.apache.org/licenses/LICENSE-2.0 for license information.
+// SPDX short identifier: Apache-2.0
+//
+//+++++++++++-+-+--+----- --- -- -  -  -   -
+
+#include <llfs/track_fds.hpp>
+//
+
+#include <llfs/logging.hpp>
+
+#include <batteries/env.hpp>
+#include <batteries/suppress.hpp>
+
+#include <filesystem>
+#include <memory>
+#include <mutex>
+#include <unordered_map>
+
+#ifdef __linux__
+//
+#include <dirent.h>
+#include <sys/types.h>
+//
+#endif
+
+namespace llfs {
+
+namespace {
+
+const char* kVarName = "LLFS_TRACK_FDS";
+
+struct TrackFdState {
+  std::atomic<bool> enabled{batt::getenv_as<int>(kVarName).value_or(0) != 0};
+  std::mutex mutex;
+  std::unordered_map<int, std::unique_ptr<boost::stacktrace::stacktrace>> traces;
+
+  //+++++++++++-+-+--+----- --- -- -  -  -   -
+
+  static TrackFdState& instance() noexcept
+  {
+    // Intentionally leaked.
```

Comment:
Explain why we're intentionally leaking, especially since we never delete.

Reply:
I've added more to the comment; the short answer is that it's usually better (IMO) to leak process-global stuff like this since running a(nother) destructor at exit only introduces the possibility of things going wrong.